lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -400,6 +400,9 @@ static void addPerfEarlyModulePassPipeline(SILPassPipelinePlan &P) {
   // Cleanup after SILGen: remove unneeded borrows/copies.
   P.addSemanticARCOpts();
 
+  P.addMem2Reg();
+  P.addSemanticARCOpts();
```

Comment:
This semantics ARC opt seems to really help clean up after `mem2reg`. I'll look into moving some of those optimizations into mem2reg so that we don't have to duplicate this pass.